bin/strata-cli/src/signet/backend.rs

```diff
@@ -26,26 +26,40 @@ use tokio::sync::mpsc::UnboundedSender;
 
 use super::EsploraClient;
 
+#[macro_export]
 macro_rules! boxed_err {
     ($name:ident) => {
         impl std::ops::Deref for $name {
             type Target = BoxedInner;
-
             fn deref(&self) -> &Self::Target {
                 self.0.as_ref()
             }
         }
-
         impl From<BoxedErr> for $name {
             fn from(err: BoxedErr) -> Self {
                 Self(err)
             }
         }
+
+        impl std::fmt::Display for $name {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                write!(f, "{:?}", self.0)
```

Comment:
Should use `{}` formatting in here since it's a `Display` impl no?

But actually, since we're just calling out to the inner type, it's slightly better to do `std::fmt::Display::fmt(&self.0, f)` (or `Debug`) since it avoids the macro expansion and any indirection for the formatter that would happen there.

Reply:
It is not good to do `Debug` stuff, i.e. `{:?}` in `Display` impls